### DISCUSSION TITLE
Setup on OpenVpn installed on Ubuntu

### DISCUSSION THREAD

**Discussion by racing57** [bnhf/openvpn-admin-plus#59] [General]
as the title really, is is possible to install this to work with an openvpn server already installed on ubuntu server vm.  i may even move this up to Azure once i have it tested etc.

thanks in advance

**Comment by bnhf** [comment]
@racing57 

You'd need to install Docker in the VM, but that's fairly trivial using their "convenience" script. After that, install Portainer and you'll be able to spin-up this project from there. Let me know if need additional guidance on either of those.

**Reply by racing57** [reply]
thanks for the swift reply, i will try that this afternoon, and see how i get on.  

thanks again :-)

**Reply by racing57** [reply]
@bnhf 

so i have the portainer installed and running on the vm, and i can access it from my machine.

this is all new to me, so please bear with me...   i have tried to add the stack as per the github page, see below

![image](https://github.com/bnhf/openvpn-admin-plus/assets/10221881/b779345b-69f1-4cda-91fd-3e8f83c06cf9)

but i get this error 

![image](https://github.com/bnhf/openvpn-admin-plus/assets/10221881/98f1f1f4-061f-42cb-bd4d-893fdeaf71d3)

as i say, i am trying to use an existing openvpn installation, and it is running on ubuntu server 22.04

any help would be greatly appreciated.  thanks

**Comment by bnhf** [comment]
@racing57 

So in the environment variables section you need to put in your actual values. The country code needs to be the two character code for your country. Then the others are fairly self explanatory, with the twist in your case being the PIVPN_SERVER value. This project is oriented towards installations that were done with PiVPN, but it will work other OpenVPN installations too.

You need to look in your /etc/openvpn directory and figure out the name that was used for your server. Assuming your server config file is server.conf, look in that file and see what the names are for your cert and key. The part before the .crt and .key should be your server name.

The environment section should look like this, with your personal values added for each:
 
```
COUNTRY=
PROVINCE=
CITY=
ORG=
EMAIL=
OU=
PIVPN_SERVER=
PIVPN_CONF=
TZ=
```

**Reply by racing57** [reply]
ahh ok, that makes sense !  i will try that

**Reply by racing57** [reply]
ok, progress :-)

just this error now

![image](https://github.com/bnhf/openvpn-admin-plus/assets/10221881/6d81bed8-eff2-4b97-a713-6955bab95516)

**Reply by bnhf** [reply]
Looking at your original screenshot, it appears you missed the "docs/" part of "docs/docker-compose.yml" in the compose field.

**Comment by racing57** [comment]
ahhhh, yes, that has now deployed...

so is there anything else i need to add, or can i open that up now ?

**Reply by bnhf** [reply]
You'll want to run through the Configuration - OpenVPN config first, to get your current settings into the database. Have your server.conf file open in another window, and you can copy and paste anything that needs to be entered.

**Reply by racing57** [reply]
ok, i will get that done now.

**Reply by racing57** [reply]
sorry, but i cannot see where i need to add these.. apologies for all the questions, i know this is going to seriously help me manage the vpn server as the estate of pc's grows, but, i am not sure where or what i need to carry across, i can see the details from where i added the variables before, but cannot see anywhere else where i need to add them.  do i have to fire this up and log into it, and do it from there ?

**Reply by bnhf** [reply]
Assuming the openvpn-admin-plus container is up-and-running, you can access it at http://[hostname of server]:8080. Login using the default credentials, and proceed to the Configuration - OpenVPN config dropdown.

**Reply by bnhf** [reply]
There are some screenshots in the README that should help.

**Comment by bnhf** [comment]
@racing57 

Did you get setup OK with openvpn-admin-plus?
